src/compas_timber/connections/t_butt.py

```diff
@@ -53,36 +83,57 @@ def add_extensions(self):
+            cutting_plane = self.cross_beam.ref_sides[self.cross_beam_ref_side_index]
+            cutting_plane.translate(-cutting_plane.normal * self.mill_depth)
             start_main, end_main = self.main_beam.extension_to_plane(cutting_plane)
         except AttributeError as ae:
             raise BeamJoinningError(beams=self.beams, joint=self, debug_info=str(ae), debug_geometries=[cutting_plane])
         except Exception as ex:
             raise BeamJoinningError(beams=self.beams, joint=self, debug_info=str(ex))
         extension_tolerance = 0.01  # TODO: this should be proportional to the unit used
-        self.main_beam.add_blank_extension(start_main + extension_tolerance, end_main + extension_tolerance, self.guid)
+        self.main_beam.add_blank_extension(
+            start_main + extension_tolerance,
+            end_main + extension_tolerance,
+            self.guid,
+        )
 
     def add_features(self):
-        """Adds the trimming plane to the main beam (no features for the cross beam).
+        """Adds the required extension and trimming features to both beams.
 
         This method is automatically called when joint is created by the call to `Joint.create()`.
 
         """
-        assert self.main_beam and self.cross_beam  # should never happen
+        assert self.main_beam and self.cross_beam
 
         if self.features:
             self.main_beam.remove_features(self.features)
+            self.cross_beam.remove_features(self.features)
 
-        cutting_plane = None
-        try:
-            cutting_plane = self.get_main_cutting_plane()[0]
-        except AttributeError as ae:
-            raise BeamJoinningError(beams=self.beams, joint=self, debug_info=str(ae), debug_geometries=[cutting_plane])
-        except Exception as ex:
-            raise BeamJoinningError(beams=self.beams, joint=self, debug_info=str(ex))
+        # get the cutting plane for the main beam
+        cutting_plane = self.cross_beam.ref_sides[self.cross_beam_ref_side_index]
+        cutting_plane.xaxis = -cutting_plane.xaxis
+        if self.mill_depth:
+            cutting_plane.translate(cutting_plane.normal * self.mill_depth)
 
-        trim_feature = CutFeature(cutting_plane)
+        # apply the cut on the main beam
+        main_feature = JackRafterCut.from_plane_and_beam(cutting_plane, self.main_beam, self.main_beam_ref_side_index)
+        self.main_beam.add_features(main_feature)
+        # store the feature
+        self.features = [main_feature]
+
+        # apply the pocket on the cross beam
         if self.mill_depth:
-            self.cross_beam.add_features(MillVolume(self.subtraction_volume()))
-        self.main_beam.add_features(trim_feature)
-        self.features = [trim_feature]
+            cross_cutting_planes = (
+                self.main_beam.ref_sides[self.main_beam_ref_side_index],
+                self.main_beam.ref_sides[(self.main_beam_ref_side_index + 2) % 4],
```

Comment:
is there a way to make sense of these values to help someone reading this code?
is this an opposing face or maybe adjacent? can `Beam.OPPOSING_SIDE_MAP` maybe be used?